fluentlenium-core/src/main/java/org/fluentlenium/core/FluentPage.java

```diff
@@ -115,7 +113,7 @@ protected void isAtUsingSelector(By by) {
     }
 
     @Override
-    public final P go() {
+    public <P extends FluentPage> P go() {
```

Comment:
I realised that I took off the `final` modifier here, but couldn't see a good reason to prevent people overriding the method (especially as the other `go` method doesn't have it) so left it off.  Can easily add back on!

Reply:
It can be removed so user has the choice to override the method in `IndexPage` subclass to return IndexPage, or call `page.<IndexPage>go()`.